instructor/common.py

```diff
@@ -0,0 +1,604 @@
+from django.shortcuts import render_to_response, redirect
+from django.template import RequestContext
+from django import forms
+from django.forms.models import modelformset_factory, inlineformset_factory
+from django.forms.models import modelform_factory
+from instructor import models
+from instructor import compiler
+from django.contrib.auth.models import User
+import datetime
+
+from backend.models import Assignment, StudentAssignment
+
+
+def courses(request):
+    user = request.user
+    message = ''
+    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'], can_delete=True)
+    if request.method == "POST":
+        formset = CourseFormSet(request.POST)
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.owner = request.user
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/courses_new.html',
+                              {'formset': CourseFormSet(queryset=models.Course.objects.filter(owner=user)),
+                               'message': message},
+                              context_instance=RequestContext(request))
+
+
+def course_delete(request, pk):
+    print "DELETE {}".format(pk)
+    models.Course.objects.get(id=pk).delete()
+    return redirect('common_course')
+
+
+def get_account_type(request):
+    account_type = ''
+    if request.user.id and len(request.user.groups.all()) > 0:
+        account_type = request.user.groups.all()[0].name
+    return account_type
+
+
+def assignments(request):
+
+    account_type = get_account_type(request)
+    public_list = models.Assignment.objects.filter(access='Public')
+
+    return render_to_response('instructor/assignments.html',
+                              {'assignments': public_list},
+                              context_instance=RequestContext(request))
+
+
+def assignments_delete(request, pk):
+    models.Assignment.objects.get(id=pk).delete()
```

Comment:
You will want to wrap this in a try:...except in case the model doesn't exist it will raise a 500, but you probably will want to return a 404 if it doesn't exist